Makefile

```diff
@@ -0,0 +1,15 @@
+all: test
```

Comment:
I can remove the Makefile from this PR so that we don't need to bikeshed it here. Give the Makefile a try, you'll be surprised how nice it makes the dev process and how it actually documents what is done at each step.